src/main/java/de/skuzzle/enforcer/restrictimports/rule/RestrictImports.java

```diff
@@ -112,11 +114,15 @@ private AnalyzerSettings createAnalyzerSettingsFromPluginConfiguration(
         if (!(includeCompileCode || includeTestCode)) {
             throw new IllegalArgumentException("Configuration error: No sources were included");
         }
+
+        final List<String> excludedSourceRoots = excludedSourceRoot != null
+                ? Collections.singletonList(excludedSourceRoot)
+                : this.excludedSourceRoots; 
         final Collection<Path> srcDirectories = this.includeCompileCode
-                ? listSourceRoots(mavenProject.getCompileSourceRoots())
+                ? listSourceRoots(filterSourceRoots(mavenProject.getCompileSourceRoots(), excludedSourceRoots))
```

Comment:
To simplify this a bit, I'd suggest passing the `excludedSourceRoots` right into the `listSourceRoots` method and do the filtering right there.
Also currently, the `LOGGER.debug` statement in `listSourceRoots` would output misleading information

Reply:
Sure. I misunderstood that debug log, I thought it was about what's actually included as source dir.

Reply:
Fixed this and also added another debug log to explain why a source root was excluded.